### PR TITLE
[CSL-1647] Add support for item_id to retrieveVariations function

### PR DESCRIPTION
**Pull request by esezen**
While adding support for `item_id` I found issues with my initial implementation. This PR fixed those issues as well.

### CODE REVIEW

File: constructor_io/modules/catalog.py [hunk 1]
```diff
@@ -53,6 +53,12 @@ def _create_query_params_for_items(parameters):
         section = parameters.get('section')
         notification_email = parameters.get('notification_email')
         force = parameters.get('force')
+        num_results_per_page = parameters.get('num_results_per_page')
```
Comment (jjl014) [inline constructor_io/modules/catalog.py:56]: Nice refactor!

File: constructor_io/modules/catalog.py [hunk 2]
```diff
@@ -413,25 +416,21 @@ def retrieve_variations(self, parameters=None):
         '''
         Retrieves multiple variations from the index (limit of 1,000)
 
-        :param list parameters.ids: A list of item IDs to retrieve
-        :param str parameters.section: The section to update
-        :param int parameters.num_results_per_page: The number of variations to return. Defaults to 100. Maximum value 100
+        :param list parameters.ids: A list of variation IDs to retrieve
+        :param str parameters.item_id: Item ID of variations to retrieve
+        :param str parameters.section: The section to retrieve from
+        :param int parameters.num_results_per_page: The number of variations to return. Defaults to 100. Maximum value 1000
```
Comment (jjl014) [inline constructor_io/modules/catalog.py:422]: Same as above, it says 100 in the [docs](https://docs.constructor.io/rest_api/items/variations/#url-parameters-2) 🤔 Are the docs incorrect?

File: constructor_io/modules/catalog.py [hunk 3]
```diff
@@ -288,8 +300,8 @@ def retrieve_items(self, parameters=None):
         Retrieves multiple items from the index (limit of 1,000)
 
         :param list parameters.ids: A list of item IDs to retrieve
-        :param str parameters.section: The section to update
-        :param int parameters.num_results_per_page: The number of items to return. Defaults to 100. Maximum value 100
+        :param str parameters.section: The section to retrieve from
+        :param int parameters.num_results_per_page: The number of items to return. Defaults to 100. Maximum value 1000
```
Comment (jjl014) [inline constructor_io/modules/catalog.py:304]: [Docs](https://docs.constructor.io/rest_api/items/items/#url-parameters-2) incorrect? 😂 Says 100

**Review by jjl014 — APPROVED** [review]
Nice, looking good to me!

Left some minor comments.

File: constructor_io/modules/catalog.py (hunk 2 above)
Comment (esezen) [reply constructor_io/modules/catalog.py:422]: Nope I just read the wrong line 😂

**Pull request merged**